File: VAMPObjCSample/VAMPObjCSample/SwitchCell.m
//
//  SwitchCell.m
//  VAMPObjCSample
//
//  Created by Supership Inc. on 2023/01/30.
//  Copyright © 2023 Supership Inc. All rights reserved.
//

#import "SwitchCell.h"

@implementation SwitchCell

- (IBAction) valueChanged:(id)sender {
    if ([self.delegate respondsToSelector:@selector(switchCell:didValueChange:)]) {
        [self.delegate switchCell:self didValueChange:self.uiSwitch.isOn];
    }
}

@end

File: VAMPSwiftSample/VAMPSwiftSample/SwitchCell.swift
//
//  SwitchCell.swift
//  VAMPSwiftSample
//
//  Created by Supership Inc. on 2023/01/30.
//  Copyright © 2023 Supership Inc. All rights reserved.
//

import UIKit

class SwitchCell: UITableViewCell {
    @IBOutlet var label: UILabel!
    @IBOutlet var uiSwitch: UISwitch!
    weak var delegate: SwitchCellDelegate?

    @IBAction func valueChanged(_ sender: Any) {
        delegate?.switchCell(self, didValueChange: uiSwitch.isOn)
    }
}

protocol SwitchCellDelegate: AnyObject {
    func switchCell(_ switchCell: SwitchCell, didValueChange value: Bool)
}
